spectrochempy/core/dataset/coord.py

```diff
@@ -178,26 +185,49 @@ def reversed(self):
         # Return a correct result only if the data are sorted  # return  # bool(self.data[0] > self.data[-1])
 
     @property
+    @_docstring.dedent
     def data(self):
+        """%(data)s
+
+        Notes
+        -----
+        The data are always returned as a 1D array of float rounded to the number
+        of significant digits given by the `sigdigits` parameters.
+        If the spacing between the data is constant with the accuracy given by the
+        significant digits, the data are thus linearized
+        and the `linear` attribute is set to True.
         """
-        The `data` array (`~numpy.ndarray`).
-
-        If there is no data but labels, then the labels are returned instead of data.
-        """
-        if self.linear:
-            data = np.arange(self.size) * self._increment + self._offset
-            # if hasattr(data, "units"):
-            #    data = data.m
-        else:
-            data = self._data
-
-        return data
+        return super().data
 
     @data.setter
     def data(self, data):
-
+        # set the data
         self._set_data(data)
 
+        # check if data is 1D
+        if self.has_data and len(self.shape) > 1:
+            raise ValueError("Only one 1D arrays can be used to define coordinates")
+
+        # linearize the data if possible or at least round it
+        # to the number of significant digits
+        if self.has_data and self.dtype.kind not in "M":
+
+            # First try to linearize the data if it is not a datetime
+            self._linear = False
+            self.linearize(self._sigdigits)
+            if self._linear:
+                # the data has been linearized, that's it!
+                return
+
+            # well, the data cannot be linearized with the given significant digits,
+            # at least round the data to the number of significant digits
+            if self._data.size < 1:
+                nd = self.sigdigits + 1
+            else:
+                maxval = np.max(np.abs(self._data))
+                nd = get_n_decimals(maxval, self.sigdigits) if maxval > 0 else 2
+            self._data = np.around(self._data, max(nd, 2))
+
```

Comment:
Arnaud, I removed all references to LinearCoord, it seems to work. 

But I have a question for you about this piece of code. 
When creating the coordinates, I do an automatic linearization (see the `linearize` function). If it is successful the `_linear` attribute is set to True, and can therefore be used to test when the coordinates must be evenly spaced (like for the FFT - interferogram/NMR).  If the linearization cannot take place `linear` is set to False. In any case, there is no more notion of offset, increment and size which were used in LinearCoord. Everything stays in the `_data` attribute. 

I added at the time of the creation, as you can see in this code snippet above,, a rounding on the stored data because it seems to me (but maybe you have a different opinion) that we don't need for coordinates to have a too important number of decimals/ significant digit. And so we can accept a rounding. For this I use the number of significant digits, which I have set to 4 by default. You can change this default of course, and you can fix it when defining Coord, since it is now a parameter. 

What do you think about it. With this process the comparison of coordinates for alignment for example are much easier I think. 

Should I increase the number of significant digits by default, which will lead to changes in some tests but it's not too complicated to change. For me 4 digits is more than enough in most cases in IR or NMR.

Reply:
@atravert

Reply:
In fact, to be more a little more correct, the number of significant digits is used but also the number of decimals which in all cases is not less than 2 (so the big numbers have all their significant digits even if > 4, but the small ones have at least 4 significant digits.

Reply:
UPDATE: it appears that rounding is sometimes a problem (such for lambdas coordinate in IRIS). I'll add the possibility of not rounding if necessary.